src/simplewallet/simplewallet.cpp

```diff
@@ -4781,7 +4781,7 @@ bool simple_wallet::donate(const std::vector<std::string> &args_)
   local_args.push_back(amount_str);
   if (!payment_id_str.empty())
     local_args.push_back(payment_id_str);
-  message_writer() << tr("Donating ") << amount_str << " to The Monero Project (donate.getmonero.org or "<< MONERO_DONATION_ADDR <<").";
+  message_writer() << tr("Donating ") << amount_str << tr(" to The Monero Project (donate.getmonero.org or ") << MONERO_DONATION_ADDR <<").";
```

Comment:
Constructing translations by concatenating sentence parts might work in some languages, however presumes a grammar and defined sentence structure.
In German for example, the proper translation would be `$AMOUNT XMR werden an das Monero Project ($URL or $ADDR) gespendet.` which has a completely different structure. (BTW: `XMR` is missing in the English text)

Rather than fixing the old and broken scheme, I suggest using formatted strings instead:

`sprintf(buf, "Donating %s XMR to the Monero Project (%s or %s).", amount_str, MONERO_DONATION_URL, MONERO_DONATION_ADDR);`

Reply:
Good point. I’d suggest the use of `boost::format` which is commonly used in the codebase.

Reply:
@leonklingele You're right, I hadn't thought of this before. Thanks